dask_cuda/local_cuda_cluster.py

```diff
@@ -39,96 +39,95 @@ def __init__(self, *args, **kwargs):
 class LocalCUDACluster(LocalCluster):
     """A variant of ``dask.distributed.LocalCluster`` that uses one GPU per process.
 
-    This assigns a different ``CUDA_VISIBLE_DEVICES`` environment variable to each
+    This assigns a different ``CUDA_VISIBLE_DEVICES`` environment variable to each Dask
     worker process.
 
-    For machines with a complex architecture mapping CPUs, GPUs, and network
-    hardware, such as NVIDIA DGX-1 and DGX-2, this class creates a local
-    cluster that tries to respect this hardware as much as possible.
+    For machines with a complex architecture mapping CPUs, GPUs, and network hardware,
+    such as NVIDIA DGX-1 and DGX-2, this class creates a local cluster that tries to
+    respect this hardware as much as possible.
 
-    It creates one Dask worker process per GPU, and assigns each worker process
-    the correct CPU cores and Network interface cards to maximize performance.
-    If UCX and UCX-Py are also available, it's possible to use InfiniBand and
-    NVLink connections for optimal data transfer performance.
+    Each worker process is automatically assigned the correct CPU cores and network
+    interface cards to maximize performance. If UCX and UCX-Py are available, InfiniBand
+    and NVLink connections can be used to optimize data transfer performance.
 
     Parameters
     ----------
-    CUDA_VISIBLE_DEVICES: str or list
-        String or list ``"0,1,2,3"`` or ``[0, 1, 2, 3]`` to restrict activity to
-        different GPUs.
-    device_memory_limit: int, float or str
-        Specifies the size of the CUDA device LRU cache, which is used to
-        determine when the worker starts spilling to host memory.  This can be
-        a float (fraction of total device memory), an integer (bytes), a string
-        (like ``"5GB"`` or ``"5000M"``), or ``"auto"``, ``0``, or ``None`` to disable
-        spilling to host (i.e., allow full device memory usage). Default is ``0.8``,
-        80% of the worker's total device memory.
-    interface: str
-        The external interface used to connect to the scheduler, usually
-        an ethernet interface is used for connection, and not an InfiniBand
-        interface (if one is available).
-    threads_per_worker: int
-        Number of threads to be used for each CUDA worker process.
-    protocol: str
-        Protocol to use for communication, e.g., ``"tcp"`` or ``"ucx"``.
-    enable_tcp_over_ucx: bool
-        Set environment variables to enable TCP over UCX, even if InfiniBand
-        and NVLink are not supported or disabled.
-    enable_infiniband: bool
-        Set environment variables to enable UCX InfiniBand support, requires
+    CUDA_VISIBLE_DEVICES : str or list of int, optional
```

Comment:
I was working off of the [style guide from numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html#sections), but I realize that it would probably be good for a user to know if the default value for a kwarg is `None`. This was mostly done to create a consistent location for the default values to be displayed, as before some kwargs didn't have their defaults displayed in the docstring.

I'm happy to change this so that `optional` kwargs now have a `default None`, but in that case I think for sake of completeness it would be good to specify that `None` is a valid argument for these kwargs, and say specifically what happens. Using `jit_unspill` as an example:

```
    jit_unspill : bool, optional
        Enable just-in-time unspilling. This is experimental and doesn't support memory
        spilling to disk. See ``proxy_object.ProxyObject`` and
        ``proxify_host_file.ProxifyHostFile`` for more info.
``` 

should become:

```
    jit_unspill : bool or None, default None
        Enable just-in-time unspilling. Can be a boolean, or None to fall back on the
        value of ``dask.jit-unspill`` in the local Dask configuration (disabling if this
        is not set). This is experimental and doesn't support memory spilling to disk.
        See ``proxy_object.ProxyObject`` and
        ``proxify_host_file.ProxifyHostFile`` for more info.
```

Does that sound good?

Reply:
IMO, I would just not document defaults and let users instead look at the function signature, although this is perhaps not the most popular of opinions. My concern mostly comes to: either we should document _everything_ or _nothing_, and over time it's generally the case that documentation tends to drift apart from implementation quite easily. In some cases we may we elaborate in the description what is the default for more complex kwargs (e.g., `device_memory_limit` and `ucx_net_devices`), and I would be fine with that.

Summarizing, I don't mind adding "optional" or "default ...", but then it should be consistently done to _all_ `LocalCUDACluster` kwargs in this PR, and hopefully extend that to the rest of dask-cuda in the future.

Reply:
These are definitely the most important/higher-level ones and IMO suffices for now. If there's more, I'd prefer that in a follow-up PR as this is getting quite extensive already. 🙂